### PR TITLE
class library: fix String -asSecs

### PR DESCRIPTION
**Pull request by dyfer**
This corrects the behavior of ```String -asSecs``` to expect the format ```hh:mm:ss.sss```. This is consistent with the documentation and the reverse method ```SimpleNumber -asTimeString```.
Fixes https://github.com/supercollider/supercollider/issues/3818.

Before merging this, I have one question:
The original implementation warns when number of seconds/minutes/hours exceeds respectively 60/60/24, and _clips these values to their maximum_. IMO the clipping should not happen and I removed that in this PR. 
So previously
```"62:1".asSecs; //warns and returns 60.1```
and now
```"62.1".asSecs; //warns and returns 62.1```

What do you think should be the right behavior?

### CODE REVIEW

**Comment by muellmusik** [comment]
I think warns and returns correct value seems sensible.

**Comment by dyfer** [comment]
Thanks for the comment @muellmusik.
I removed the commented out lines and left the warning message, returning the correct ("out of range") value.
In the process I also realized, that for (unsupported) negative numbers within the string (like ```"02:-50"```) the warning was also posted, but the number itself (seconds, it this case) was zeroed out. I changed it so it uses an absolute value for the given number, in addition to warning that this is not supported. Is this OK?

**Comment by dyfer** [comment]
I found one more thing:
In ```SimpleNumber -asTimeString``` fractions were not handled correctly (because we were padding a float instead of an integer). The second commit fixes that.
Previously:
```123.002.asTimeString; // 00:02:03.2.0```
And now:
```123.002.asTimeString; // 00:02:03.002```

File: SCClassLibrary/Common/Collections/String.sc [hunk 1]
```diff
@@ -505,24 +505,23 @@ String[char] : RawArray {
...
-			};
+			num = num.asFloat;
 			if (num < 0) {
-				("asSecs: negative numbers within slots not supported:" + this).warn;
-				num = 0;
+				format("%.asSecs: negative numbers within slots not supported, using absolute value", this).warn;
+				num = num.abs;
+			};
+			if (num > limits[i]) {
+				format("%.asSecs: number of % greater than %, continuing", this, slotNames[i], limits[i]).warn;
```
Comment (brianlheim) [inline SCClassLibrary/Common/Collections/String.sc]: a warning implies that the function continues executing, "continuing" can be removed

**Comment by dyfer** [comment]
Thanks @brianlheim, I made the requested change.
I also added some tests, do they look OK? I haven't created tests before...

**Comment by brianlheim** [comment]
Can also confirm that the tests pass.

**Comment by brianlheim** [comment]
Thanks again @dyfer !

**Comment by blacksound** [comment]
Dear devs,
This commit broke something in my code. Since the methods `.asSecs`/`.asTimeString` are said to be reverse of eachother, I propose that the test methods should reflect this.
Her are two randomised test snubs. Before this commit these tests fails sometimes, so there is clearly a bug in there somewhere. But after this commit none of these test pass anymore.
```
(
//from random time units padded unit strings
var minutes, seconds, hours, millis, days;
var timestring;
var secs, secondTimestring;
days = rrand(0, 355);
hours = rrand(0, 23);
minutes = rrand(0, 59);
seconds = rrand(0, 59);
millis = rrand(0, 999);
timestring = "%:%:%:%.%".format(
	days.asString.padLeft(3, "0"),
	hours.asString.padLeft(2, "0"),
	minutes.asString.padLeft(2, "0"),
	seconds.asString.padLeft(2, "0"),
	millis.asString.padLeft(3, "0")
);
secs = timestring.asSecs;
secondTimestring = secs.asTimeString;
"Secs: '%' passed: %\n\t1.timestring: '%'\n\t2.timestring: '%'".format(
	secs, timestring == secondTimestring , timestring, secondTimestring).postln;
)

(
var targetSecs, resultSecs;
var timestring;
targetSecs = rand(60 * 60 * 24) / 1000.0;
timestring = targetSecs.asTimeString;
resultSecs = timestring.asSecs;

"Target secs: % Result secs: % passed: %".format(targetSecs, resultSecs, targetSecs == resultSecs).postln;
)
```

**Comment by blacksound** [comment]
This test snub shows that the version before this commit fails on a few of the tests, and that all failures probably has to something to with the overflow of seconds, as the failure values are in the range 60.0 – 90.0.

```
(
var successes=0, failures=[];
10000.do{|i|
	var targetSecs, resultSecs;
	var timestring;
	targetSecs = rand(60 * 60 * 24) / 1000.0;
	timestring = targetSecs.asTimeString;
	resultSecs = timestring.asSecs;
	if(targetSecs != resultSecs, {
		failures = failures.add(targetSecs);
	}, {
		successes = successes + 1;
	});
	// "Target secs: % Result secs: % passed: %".format(targetSecs, resultSecs, targetSecs == resultSecs).postln;
};
"Successes: % failures: %".format(successes, failures.size).postln;
f = failures;
)
```

**Comment by brianlheim** [comment]
Hi @blacksound, please open a new issue ticket with a simple reproducer.

**Comment by dyfer** [comment]
@brianlheim actually it seems this PR didn't get included in 3.10 for some reason!
Compare: [develop](https://github.com/supercollider/supercollider/blob/c33d92e537e8a7fccddbca24c7a2ea4c12973a88/SCClassLibrary/Common/Collections/String.sc#L507) (correct) vs [3.10 ](https://github.com/supercollider/supercollider/blob/9bec036558a5dbdffcd2bd4aec3de49800a80157/SCClassLibrary/Common/Collections/String.sc#L506) branch.

@blacksound feel free to check if your String.sc actually includes this fix. I tested your code on a branch that actually includes this fix and it seem to pass every time (run a dozen time or so).

**Comment by dyfer** [comment]
@brianlheim I tried creating (my first) cherry-pick PR, is that the way to do it?

**Pull request merged**